storage/storage_dest.go

```diff
@@ -328,6 +328,14 @@ func (s *storageImageDestination) tryReusingBlobAsPending(digest digest.Digest,
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
+	toc, err := chunked.GetTOCDigest(options.Annotations)
+	if err != nil {
+		return false, private.ReusedBlob{}, err
+	}
+	if toc != nil {
+		digest = *toc
+	}
```

Comment:
FIXME: `GetTOCDigest` can return two different values. Why isn’t that ambiguity a risk? It seems that we read the same data structure in both cases, so that seems fine — look into that more.

Reply:
`GetTOCDigest` still accepts images with both, but `GetDiffer` rejects images with both TOC annotations after https://github.com/containers/storage/pull/1778. So if we ever find a c/storage layer (to match), we know that TOC digest matches the contents.